control-plane/connect-inject/webhookv2/mesh_webhook.go

```diff
@@ -251,6 +252,16 @@ func (w *MeshWebhook) Handle(ctx context.Context, req admission.Request) admissi
 
 	w.Log.Info("received pod", "name", req.Name, "ns", req.Namespace)
 
+	// Validate that none of the pod ports start with the prefix "cslport-" as that may result in conflicts with ports
+	// created by the pod controller when creating workloads.
+	for _, c := range pod.Spec.Containers {
+		for _, p := range c.Ports {
+			if strings.HasPrefix(p.Name, constants.UnnamedWorkloadPortNamePrefix) {
```

Comment:
not necessarily for this PR but are we tracking anywhere restrictions like this that we should document for end users?

Reply:
For this particular case I'd not expect users to run into this and would expect if they do that it's easily discoverable through the pod not being accepted for injection. It definitely feels like a thing that'd be glazed over in any docs. Any other restrictions (so far) are enforced through kube validations